### PR TITLE
Feature/UI login

### PR DESCRIPTION
**Pull request by Malak67**
Added Login page

### CODE REVIEW

File: apps/drec-ui/src/api/user.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    const clearUser = () => {
+        const token = getAuthenticationToken();
+        if (token) {
+            removeAuthenticationToken();
+            queryClient.removeQueries(userQueryKey);
+            navigate('/');
+            queryClient.resetQueries();
+        }
+    };
+    console.log('tokenExists: ', Boolean(getAuthenticationToken()));
```
Comment (alexworker23) [inline apps/drec-ui/src/api/user.ts:68]: console.log rm

File: apps/drec-ui/src/api/user.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    const userQueryKey = getUserControllerMeQueryKey();
+    const navigate = useNavigate();
+
+    const clearUser = () => {
+        const token = getAuthenticationToken();
+        if (token) {
+            removeAuthenticationToken();
+            queryClient.removeQueries(userQueryKey);
+            navigate('/');
+            queryClient.resetQueries();
```
Comment (alexworker23) [inline apps/drec-ui/src/api/user.ts:65]: You can use single function call `queryClient.clear()` and it will remove all queries data.
Current approach will lead to double fetch of user data

File: apps/drec-ui/src/components/layout/MainLayout/MainLayout.tsx [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { FC } from 'react';
+import { Outlet } from 'react-router-dom';
+import { PageWrapper } from '../../../core';
+
+export const MainLayout: FC = () => {
+    return (
+        <>
```
Comment (alexworker23) [inline apps/drec-ui/src/components/layout/MainLayout/MainLayout.tsx:7]: Unnecessary use of Fragment

File: apps/drec-ui/src/pages/LoginPage/LoginPage.effects.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+export const useLogInPageEffects = () => {
+    const navigate = useNavigate();
+    const dispatchModals = useUserModalsDispatch();
+    const useUserLogInFormConfig = (
+        formSubmitHandler: (values: UnpackNestedValue<TUserLoginFormValues>) => void
+    ): GenericFormProps<TUserLoginFormValues> => {
+        return {
+            buttonFullWidth: true,
+            buttonText: 'Login',
```
Comment (alexworker23) [inline apps/drec-ui/src/pages/LoginPage/LoginPage.effects.ts:27]: Do we consider using localization from the beginning ?

File: apps/drec-ui/src/pages/LoginPage/LoginPage.tsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+                <DrecLogo />
+                <GenericForm {...formProps}>
+                    {/* <Box>
+                        <Button variant="text" onClick={navigateToResetPassword}>
+                            Forgot password?
+                        </Button>
+                    </Box> */}
+                </GenericForm>
+                {/* <Box>
+                    <Typography>Don't have an account?</Typography>
```
Comment (alexworker23) [inline apps/drec-ui/src/pages/LoginPage/LoginPage.tsx:26]: If we don't need this commented code I think it could be removed

File: apps/drec-ui/src/shared/logic/loginRegisterOrg.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
+export const useLoginRegisterOrgModalLogic = (
+    closeModal: () => void,
+    navigateToRegister: () => void
+) => {
+    return {
+        title: 'Thank you for registering as a user on the marketplace!',
+        text: 'To unlock all the functionalities of the marketplace, like trading I-RECs, users also need to register an organization.',
```
Comment (alexworker23) [inline apps/drec-ui/src/shared/logic/loginRegisterOrg.ts:7]: D-RECs*

File: apps/drec-ui/src/pages/LoginPage/LoginPage.effects.ts (hunk 4 above)
Comment (Malak67) [reply apps/drec-ui/src/pages/LoginPage/LoginPage.effects.ts:27]: No localization required

File: apps/drec-ui/tsconfig.json [hunk 7]
```diff
@@ -18,7 +18,7 @@
         "resolveJsonModule": true,
         "importHelpers": true,
         "isolatedModules": true,
-        "strictNullChecks": false,
+        "strictNullChecks": true,
```
Comment (alexworker23) [inline apps/drec-ui/tsconfig.json:21]: Why did you change this rule?

Comment (Malak67) [reply apps/drec-ui/tsconfig.json:21]: To see any issue before pushing.

Comment (Malak67) [reply apps/drec-ui/tsconfig.json:21]: we have it enable for the API

Comment (alexworker23) [reply apps/drec-ui/tsconfig.json:21]: But this can break the build now. You also have to adjust all initial `null`'s throughout the UI

Comment (Malak67) [reply apps/drec-ui/tsconfig.json:21]: I`ll change it back

**Review by alexworker23 — APPROVED** [review]
